enterprise/micronaut/src/org/netbeans/modules/micronaut/db/MicronautEntity.java

```diff
@@ -0,0 +1,831 @@
+import org.openide.DialogDescriptor;
+import org.openide.DialogDisplayer;
+import org.openide.NotifyDescriptor;
+import org.openide.filesystems.FileObject;
+import org.openide.filesystems.FileUtil;
+import org.openide.loaders.DataObject;
+import org.openide.loaders.TemplateWizard;
+import org.openide.util.NbBundle;
+
+/**
+ *
+ * @author Dusan Balek
+ */
+public class MicronautEntity extends RelatedCMPWizard {
+
+    private static final String TYPE_MICRONAUT = "micronaut"; // NOI18N
+
+    public static MicronautEntity create() {
+        return new MicronautEntity(TYPE_MICRONAUT);
+    }
+
+    @NbBundle.Messages({
+        "MSG_NoDbConn=No database connection found",
+        "MSG_NoProject=No project found for {0}",
+        "MSG_NoSourceGroup=No source group found for {0}",
+        "MSG_SelectTables=Select Database Tables",
+        "MSG_NoDbTables=No database table found for {0}"
+    })
+    public static CreateFromTemplateHandler handler() {
+        return new CreateFromTemplateHandler() {
+            @Override
+            protected boolean accept(CreateDescriptor desc) {
+                return true;
+            }
+
+            @Override
+            protected List<FileObject> createFromTemplate(CreateDescriptor desc) throws IOException {
+                try {
+                    FileObject folder = desc.getTarget();
+                    Project project = FileOwnerQuery.getOwner(folder);
+                    if (project == null) {
+                        DialogDisplayer.getDefault().notifyLater(new NotifyDescriptor.Message(Bundle.MSG_NoProject(folder.getPath()), NotifyDescriptor.ERROR_MESSAGE));
+                        return Collections.emptyList();
+                    }
+                    SourceGroup sourceGroup = SourceGroups.getFolderSourceGroup(ProjectUtils.getSources(project).getSourceGroups(JavaProjectConstants.SOURCES_TYPE_JAVA), folder);
+                    if (sourceGroup == null) {
+                        DialogDisplayer.getDefault().notifyLater(new NotifyDescriptor.Message(Bundle.MSG_NoSourceGroup(folder.getPath()), NotifyDescriptor.ERROR_MESSAGE));
+                        return Collections.emptyList();
+                    }
+                    DatabaseConnection connection = ConnectionManager.getDefault().getPreferredConnection(true);
+                    if (connection == null) {
+                        DialogDisplayer.getDefault().notifyLater(new NotifyDescriptor.Message(Bundle.MSG_NoDbConn(), NotifyDescriptor.ERROR_MESSAGE));
+                        return Collections.emptyList();
+                    }
+                    ConnectionManager.getDefault().connect(connection);
+                    Connection conn = connection.getJDBCConnection();
+                    ResultSet rs = conn.getMetaData().getTables(conn.getCatalog(), conn.getSchema(), "%", new String[]{"TABLE", "VIEW"}); //NOI18N
+                    List<NotifyDescriptor.QuickPick.Item> dbItems = new ArrayList<>();
```

Comment:
Let's ask @sdedic , but I believe `createFromTemplate(CreateDescriptor desc)` was never intended for displaying UI. I'd rather see that user-related code in a separate method which could be invoked by Swing as well as VSCode UI optionally.